recommenders/evaluation/spark_evaluation.py

```diff
@@ -630,6 +659,39 @@ def _get_cosine_similarity(self, n_partitions=200):
             )
         return self.df_cosine_similarity
 
+    @staticmethod
+    @udf(returnType=DoubleType())
+    def sim_cos(v1,v2):
+        try:
+            p = 2
+            return float(v1.dot(v2))/float(v1.norm(p)*v2.norm(p))
+        except:
+            return 0
```

Comment:
do we want to indicate that two vectors are identical if there's a failure to compute the distance? can we return a NaN or something?

Reply:
if we vectors are identical, their cosine similarity will be 1. On which situation do you think the calculation will fail? @gramhagen

Reply:
sorry you're right, we're not computing a distance here. i think in this case returning a 0 is okay in the exception. my concern is just about silent failures, if for some reason there is an exception, most likely because the vectors are not the correct size or have invalid values, then we don't ever surface the fact that one of the item vectors is incorrectly defined.

Reply:
added logic to check the size of vectors. If the sizes of two input vector are different, raise an exception

Reply:
i'm a still a bit confused on this part, instead of both catching a particular exception and raising an exception and providing a generic except that silently swallows the exception and returns a 0. I think we just do one or the other. I'm fine with using a generic exception and returning a NaN or not even using a try-except and throwing the exception.

Reply:
It's probably better practice to pass the exception through instead of returning 0, because people interpret 0 as the two vectors being orthogonal. Also passing NaN in the case of an exception is not great because people would expect NaN to be the output when both vectors are zero.

Reply:
Although the latter is not so clear in this case, e.g. if both feature vectors are zero then they may be considered similar.

Reply:
removed try/catch block